nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/PutAzureCosmosDBDocument.java

```diff
@@ -0,0 +1,145 @@
+
+        final Set<Relationship> _relationships = new HashSet<>();
+        _relationships.add(REL_SUCCESS);
+        _relationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(_relationships);
+        mapper = new ObjectMapper();
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final ComponentLog logger = getLogger();
+
+        final Charset charset = Charset.forName(context.getProperty(CHARACTER_SET).getValue());
+        final String partitionKeyField = context.getProperty(PARTITION_KEY).getValue();
+        JsonNode doc = null;
+
+        try {
+            // Read the contents of the FlowFile into a byte array
+            final byte[] content = new byte[(int) flowFile.getSize()];
+            session.read(flowFile, in -> StreamUtils.fillBuffer(in, content, true));
+
+            // parse conten into JsonNode object
+            doc = mapper.readTree(new String(content, charset));
+
+            // make sure type of id is String type if exists
+
+            if(doc.has("id")) {
+                JsonNode idNode = doc.get("id");
+                if(idNode.isNumber()) {
+                    logger.debug("coverting number id into string...");
+                    ((ObjectNode) doc).put("id", doc.get("id").asText());
+                }
+            } else {
+                ((ObjectNode) doc).put("id",flowFile.getAttribute("uuid"));
+            }
+            logger.debug("after document ID verification");
+            if(!doc.has(partitionKeyField)){
+                logger.error("Required parition key field value not found from input flow file content with doc.id=" + doc.get("id"));
+                session.transfer(flowFile, REL_FAILURE);
+                context.yield();
+            } else {
+                // insert doc if this is a new, otherwise update the existing document with upsertDocument API
+                logger.debug("Inserting/Updating document with upsertItem");
+                this.container.upsertItem(doc);
+
+                session.getProvenanceReporter().send(flowFile, getURI(context));
+                session.transfer(flowFile, REL_SUCCESS);
+                logger.debug("Inserting/Updating document completed...");
+            }
+        } catch (Exception e) {
+            logger.error("Failed to upsertDocument {} into CosmosDB due to {}", new Object[] {flowFile, e}, e);
+            if(doc != null){
+                logger.error("Failed doc content : " + doc.toPrettyString());
```

Comment:
The max size for these can be big, right? I wouldn't really want to dump a couple of MB. Maybe a debug or user can snag it out of provenance if needed.

Reply:
wrapped with logger.isDebugEnabled().

Reply:
You wrapped it with ``logger.isErrorEnabled()`` but that's not what @jfrazee is suggesting. In this case, we most likely don't want to log the content of the document that we tried to send in CosmosDB as it could be a large document. Besides, as a general rule, we try to send into logs as little flow file content as possible as this is a potential way of leaking sensitive data.

I'd completely remove this log statement. If there is an error, a user can use the provenance data and still access the content of the failed flow file (assuming it's still in the archive part of the content repository). Besides it's up to the user to correctly process the failed flow files.

Reply:
removed the line of logging failed document.